x-pack/plugins/ml/public/application/trained_models/models_management/test_models/models/lang_ident/lang_ident_output.tsx

```diff
@@ -6,22 +6,32 @@
  */
 
 import React, { FC } from 'react';
+import useObservable from 'react-use/lib/useObservable';
 import { i18n } from '@kbn/i18n';
 import { EuiSpacer, EuiBasicTable, EuiTitle } from '@elastic/eui';
 
-import type { FormattedLangIdentResp } from './lang_ident_inference';
+import type { LangIdentInference } from './lang_ident_inference';
 import { getLanguage } from './lang_codes';
 
 const PROBABILITY_SIG_FIGS = 3;
 
-export const LangIdentOutput: FC<{ result: FormattedLangIdentResp }> = ({ result }) => {
-  if (result.length === 0) {
+export const getLangIdentOutputComponent = (inferrer: LangIdentInference) => (
+  <LangIdentOutput inferrer={inferrer} />
+);
+
+const LangIdentOutput: FC<{ inferrer: LangIdentInference }> = ({ inferrer }) => {
+  const result = useObservable(inferrer.inferenceResult$);
+  if (!result) {
```

Comment:
Couldn't this and the condition on line 28 be in the same block as they both return null?

Reply:
https://github.com/elastic/kibana/pull/129209/commits/b20fd1ed64ecd9c3e07427b19ad3be0a6e14a241